compiler/rustc_span/src/hygiene.rs

```diff
@@ -1163,6 +1163,9 @@ pub enum DesugaringKind {
     WhileLoop,
     /// `async Fn()` bound modifier
     BoundModifier,
+    /// Marks a `&raw const *_1` needed as part of getting the length of a mutable
+    /// slice for the bounds check, so that MIRI's retag handling can recognize it.
+    IndexBoundsCheckReborrow,
```

Comment:
Span should *definitely* not be used like this. Span desugarings should only ever be used for diagnostics and things like that -- making miri rely on this behavior seems really bad. If we need to track certain locals for the purposes of some special miri, this needs to be recorded in the MIR body separately.

cc @scottmcm @RalfJung

Reply:
It's a pretty bad hack, yeah. Medium-term I hope that Miri can entirely stop doing reborrowing on `&raw`, but this currently runs into issues... so not sure if it's worth spending a lot of effort on the proper thing.